.buildkite/scripts/pre-install-command.sh

```diff
@@ -12,16 +33,34 @@ add_bin_path(){
     export PATH="${WORKSPACE}/bin:${PATH}"
 }
 
-with_mage() {
+with_go() {
     create_bin_folder
-    retry 5 curl -sL -o "${WORKSPACE}/bin/mage.tar.gz" "https://github.com/magefile/mage/releases/download/v${SETUP_MAGE_VERSION}/mage_${SETUP_MAGE_VERSION}_Linux-64bit.tar.gz"
+    check_platform_architecture
+
+    echo "--- Install Golang"
+    echo "GVM ${SETUP_GVM_VERSION} (platform ${platform_type_lowercase} arch ${arch_type}"
+    retry 5 curl -sL -o "${WORKSPACE}/bin/gvm" "https://github.com/andrewkroh/gvm/releases/download/${SETUP_GVM_VERSION}/gvm-${platform_type_lowercase}-${arch_type}"
+
+    chmod +x "${WORKSPACE}/bin/gvm"
+    eval "$(gvm "$(cat .go-version)")"
+    go version
+    which go
+    PATH="${PATH}:$(go env GOPATH)/bin"
+    export PATH
+}
+
+with_mage() {
+    if ! command -v go &> /dev/null ; then
+        with_go
+    fi
```

Comment:
There are steps that run in golang containres, that would not need to install go again.